saleor/graphql/core/connection.py

```diff
@@ -1,7 +1,168 @@
+import typing
+from typing import Any, Dict
+
 import graphene
+from django.db.models import Q, QuerySet
 from graphene import Field, List, NonNull, ObjectType, String
 from graphene.relay.connection import Connection
 from graphene_django_optimizer.types import OptimizedDjangoObjectType
+from graphql_relay.connection.connectiontypes import Edge, PageInfo
+from graphql_relay.utils import base64, unbase64
+
+from ..core.enums import OrderDirection
+
+ConnectionArguments = Dict[str, Any]
+
+
+def to_global_cursor(values):
+    if not isinstance(values, List):
+        values = [str(values)]
+    else:
+        values = [str(value) for value in values]
+    return base64(":".join(values))
+
+
+def from_global_cursor(cursor) -> typing.List[str]:
+    values = unbase64(cursor)
+    values = values.split(":")
+    if isinstance(values, list):
+        return values
+    return [values]
+
+
+def connection_args_validation(args):
+    first = args.get("first")
+    last = args.get("last")
+
+    if isinstance(first, int) and first < 0 or first and not isinstance(first, int):
+        raise ValueError("Argument 'first' must be a non-negative integer.")
+    if isinstance(last, int) and last < 0 or last and not isinstance(last, int):
+        raise ValueError("Argument 'last' must be a non-negative integer.")
+    if first and last:
+        raise ValueError("Argument 'last' cannot be combined with 'first'.")
+    if first and args.get("before"):
+        raise ValueError("Argument 'first' cannot be combined with 'before'.")
+    if last and args.get("after"):
+        raise ValueError("Argument 'last' cannot be combined with 'after'.")
+
+
+def connection_from_queryset_slice(
+    qs: QuerySet,
+    args: ConnectionArguments = None,
+    connection_type: Any = Connection,
+    edge_type: Any = Edge,
+    page_info_type: Any = PageInfo,
+) -> Connection:
+    """Create a connection object from a QuerySet."""
+    args = args or {}
+    before = args.get("before")
+    after = args.get("after")
+    first = args.get("first")
+    last = args.get("last")
+    connection_args_validation(args)
+
+    cursor_after = from_global_cursor(after) if after else None
+    cursor_before = from_global_cursor(before) if before else None
+    cursor = cursor_after or cursor_before
+    cursor_offset = 1 if cursor else 0
+
+    sort_by = args.get("sort_by", {})
+    sorting_direction = sort_by.get("direction", "")
```

Comment:
Shouldn't this default to `OrderDirection.ASC`?

Reply:
I need `"-"` and `""` in order_by that's why I did it this way

Reply:
What if I pass `OrderDirection.ASC` as `direction`?

Reply:
If you pass `OrderDirection` it will be parsed as `""` or `"-"` so I end up having string